explainer.md

```diff
@@ -161,119 +150,107 @@ user agent.
+badge that applies to the current app:
 
 ```js
-Badge.set(getUnreadCount(location.pathname), {scope: location});
+navigator.setAppBadge(getUnreadCount());
 ```
 
-The scope is a URL prefix; the badge is applied on all pages whose URL [starts
-with](https://www.w3.org/TR/appmanifest/#dfn-within-scope) that prefix.
-
-More advanced examples are given in a [separate document](docs/examples.md).
-
-## Badge scope
-
-This section explains the difference between badging a *set of URLs* versus
-badging a *document*.
+As above, a value of 0 clears the badge and `navigator.clearAppBadge` can also
+explicitly clear the badge for this origin. The effects of the app API are
+global and may outlast the document (it is intended to persist at least until
+the user agent closes). It can also be used from a service worker, although the
+meaning of that will be somewhat different.
 
-Consider four increasingly narrow sets of pages that a Badge is applied to:
+The reason we have separate APIs is that they have very different concerns which
+are now properly separated:
 
-1. An entire origin (e.g., when you just want to set an unread count for the
-   entire site).
-2. All pages whose URL [starts
-   with](https://www.w3.org/TR/appmanifest/#dfn-within-scope) a given path
-   (e.g., if a site is organised into "projects", each with their own status
-   indicator).
-3. All pages with a specific URL path (e.g., if each page has a separate status
-   indicator, but all tabs open at a certain URL should share status).
-4. A specific document (e.g., if many tabs can be open with URLs on the same
-   path, but might have different status, due to either a different query
-   string, or different dynamic context inside the document).
+* The push message complexity only applies to the app API.
+* The lifetime of the app API is considerably different to the document API.
+* The presence of the document API itself indicates whether the tab will be
+  badged in the UI (obviating the need for a dedicated feature detection method
+  specifically for tab badging).
 
-Number 3 and 4 seem very similar, but architecturally are very different. In
-3, the badge is associated with a URL (and any current or future documents open
-at that URL will show that badge), whereas in 4, the badge is associated with a
-particular document — other tabs open to the same URL will not share the badge.
-
-What this boils down to is two different modes:
+Other reasons to separate:
 
-* Badging a particular URL scope, applied to the set of URLs that start with the
-  same path. This covers 1, 2 and 3 above.
-* Badging a particular document, applied to only the browser tab containing the
-  document.
+* Sites that only want to badge their app icon or only want to badge their tab
+  can just concern themselves with the relevant API.
+* The document API can potentially support badges that aren't supported by host
+  operating systems, such as arbitrary Unicode characters.
```

Comment:
Should make it clear that this is currently not supported by the proposal, but a future extension reliant on implementer feedback.